support-frontend/assets/components/subscriptionCheckouts/endSummary/endSummarySelector.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import { getProductPrice } from 'helpers/productPrice/productPrices';
+import { getPriceDescription, displayPrice } from 'helpers/productPrice/priceDescriptions';
+import { extendedGlyph } from 'helpers/internationalisation/currency';
+
+import type { ProductPrice } from 'helpers/productPrice/productPrices';
+import type { CheckoutState } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
+
+const getPromotion = (productPrice: ProductPrice) =>
+  productPrice.promotions &&
+  (productPrice.promotions.length > 0
+    ? productPrice.promotions[0].description
+    : null);
+
+function mapStateToProps(state: CheckoutState) {
```

Comment:
I have a question as I'm trying to follow this and am not 100% upto speed with the use of Redux here. 

The `EndSummary` component takes the following `Props`:

```ts
// EndSummary.jsx

type Props = {
  priceDescription: string,
  promotion: Option<string>,
  displayPrice: string,
}; 
```

This `mapStateToProps` to function returns an object which matches the type signature of the `Props` above, and this `mapStateToProps` function is retrieving these values from the state in the Redux store? 

So there's a contract between the object returned by `mapStateToProps` and the `Props` the `EndSummary` component accepts? 

Therefore should `mapStateToProps` have a defined return type that matches the `Props` signature? What happens if for example we remove one of the properties from the Object returned by `mapStateToProps`, for example `priceDescription` which is a required property in the EndSummary `Props`? Would `flow` flag that the contract has been broken?

Reply:
In short, I'm thinking we should have return types defined for the functions in this file.